Classes/ValidatorExtension.swift

```diff
@@ -0,0 +1,21 @@
+public extension XCTestCase {
+    typealias ValidationFunction = (BatchElement) -> ()
+    func segmentValidator(result: Result<[BatchElement], Error>, expectation: XCTestExpectation, maxRetries: Int, currentRetries: Int, validator: @escaping ValidationFunction) {
+        switch result {
+        case .success(let finalResult):
+            XCTAssertFalse(finalResult.isEmpty)
+            finalResult.forEach(validator)
+            expectation.fulfill()
+        case .failure:
+            guard currentRetries < maxRetries else {
+                XCTFail()
+                expectation.fulfill()
+                return
+            }
+            SegmentService().checkForSegmentCalls(expectedCallType: "screen", expectedPageType: "stories") { [weak self]
```

Comment:
im assuming you don't want to hardcode these types right?

Reply:
Ah good catch. Thank you.